packages/picker/src/Picker.ts

```diff
@@ -311,7 +312,20 @@ export class PickerBase extends SizedMixin(SpectrumElement, {
     };
 
     protected async keyboardOpen(): Promise<void> {
-        this.toggle(true);
+        // if the menu is not open, we need to toggle it and wait for it to open to focus on the first selected item
+        if (!this.open || !this.strategy.open) {
```

Comment:
Opening via keyboard set focus on first item. If menu isn't already open, menu will be opened first.